stack-nodejs-16/build.Dockerfile

```diff
@@ -0,0 +1,5 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
+ENV CNB_USER_ID=1001
+ENV CNB_GROUP_ID=1000
+ENV CNB_STACK_ID="io.buildpacks.stacks.ubi8"
+ENV CNB_STACK_DESC="ubi nodejs minimal run image base"
```

Comment:
I believe these `CNB_` variables do not have an affect since `jam create-stack` picks up these values from stack.toml file. jam maintainers will have the best idea though @sophiewigmore

Reply:
Yep, you're right. Good catch, jam will set these for you 👍

Reply:
I can try again,  but I'm pretty sure I had removed these specific ones, ran into trouble and added them back.  But I could be remembering wrong so I'll check again.

Reply:
I just built with the ENV's removed from the docker file. I don't see jam adding them:

```shell

user1@fedora scripts]$ docker images
REPOSITORY                                 TAG       IMAGE ID       CREATED        SIZE
localhost:5000/ubi8-paketo-run-nodejs-16   latest    dafc5e977ba7   4 weeks ago    185MB
localhost:5000/ubi8-paketo-run-nodejs-18   latest    721414ef396d   4 weeks ago    196MB
localhost:5000/ubi8-paketo-build           latest    e541858452fd   4 weeks ago    93MB
localhost:5000/ubi8-paketo-run             latest    e541858452fd   4 weeks ago    93MB
registry                                   2         81c944c2288b   6 months ago   24.1MB
[user1@fedora scripts]$ docker run -it dafc5e977ba7 /bin/bash
bash-4.4$ env
NPM_RUN=start
SUMMARY=Minimal image for running Node.js 16 applications
HOSTNAME=e4abecb0294f
DESCRIPTION=Node.js 16 available as container is a base platform for running various Node.js 16 applications and frameworks. Node.js is a platform built on Chrome's JavaScript runtime for easily building fast, scalable network applications. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient, perfect for data-intensive real-time applications that run across distributed devices.
container=oci
PWD=/opt/app-root/src
HOME=/opt/app-root/src
NAME=nodejs
NODEJS_VERSION=16
TERM=xterm
PLATFORM=el8
SHLVL=1
APP_ROOT=/opt/app-root
PATH=/opt/app-root/src/node_modules/.bin/:/opt/app-root/src/.npm-global/bin/:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
NPM_CONFIG_PREFIX=/opt/app-root/src/.npm-global
_=/usr/bin/env

```

Reply:
@sophiewigmore can you think of anything that would be needed for them to be set that I'm not doing?  The create script that calls jam should be pretty much the same as the existing ones for the other stacks.  This is the diff

```shell
user1@fedora check2]$ !diff
diff creat2 ../ubi-base-stack/scripts/create-one.sh
8c8
< readonly STACK_DIR="${ROOT_DIR}/stack"
---
> readonly STACK_DIR="${STACK_DIR:-${ROOT_DIR}/stack}"
10c10
< readonly BUILD_DIR="${ROOT_DIR}/build"
---
> readonly BUILD_DIR="${BUILD_DIR:-${ROOT_DIR}/build}"
```

Reply:
When you run with `jam` locally, do you see log lines that say `Adding CNB_* environment variables`? it should say it right after some log about decorating the base image

Reply:
This is the output from the local run:

```shell
user1@fedora scripts]$ /home/user1/paketo/ubi-base-stack/.bin/jam create-stack --config /home/user1/paketo/ubi-base-stack/stack/stack.toml --build-output /home/user1/paketo/ubi-base-stack/build/build.oci --run-output /home/user1/paketo/ubi-base-stack/build/run.oci
Building io.buildpacks.stacks.ubi8
  Building on linux/amd64
    Building base images
      Build complete for base images
    build: Decorating base image
      Adding CNB_* environment variables
      Adding io.buildpacks.stack.* labels
      Creating cnb user
    run: Decorating base image
      Adding io.buildpacks.stack.* labels
      Creating cnb user
      Updating /etc/os-release
    build: Updating image
    run: Updating image

  Exporting build image to /home/user1/paketo/ubi-base-stack/build/build.oci
  Exporting run image to /home/user1/paketo/ubi-base-stack/build/run.oci
```

and the jam version info

```shell
[user1@fedora scripts]$ /home/user1/paketo/ubi-base-stack/.bin/jam version
jam 1.7.0
```

Reply:
I tried with a newer version of jam but believe its the same.

What I do notice is that it says adding "Adding CNB_* environment variables" for build image but not the run image

Reply:
Experimenting more and looking closer I also discovered that my generation scripts erroneously were using the run image for both the run and build image when loading into registries. 

That might explain why I had failures when tried removing them. ie:  the variables are only needed in the build image as appears might be the case from the jam output and my failures were because even though jam added them to the build image I was using the run image instead during the build step.

Testing that out now.

Reply:
@arjun024 and @sophiewigmore thanks for the help in figuring that out. Does seem to work ok with the ENV's just being added to the build image.  

I've pushed a commit to remove the extra ENVs from the Docker files and to fix my temporary script in loading them in locally.

Reply:
The run images don't get the environment variables added, just the build image, I don't think I realized that before. Inspecting our Paketo Jammy Tiny Stack, for example, the run image has no env. vars added so I think we are good to go